keps/sig-node/24-apparmor/README.md

```diff
@@ -0,0 +1,790 @@
+initial implementation. If this feature is implemented in v1.26, I propose v1.30 as a
+target for removal of the old behavior. Specifically, annotation support will be removed
+in the kubelet after this period, and fields will no longer be copied to annotations for older kubelet 
+versions. However, annotations submitted to the API server will continue to be copied to fields at the 
+kubelet indefinitely, as was done with Seccomp.
+
+### Test Plan
+
+AppArmor already has [e2e tests][https://github.com/kubernetes/kubernetes/blob/6596a14/test/e2e_node/apparmor_test.go],
+but the tests are guarded by the `[Feature:AppArmor]` tag and not run in the
+standard test suites.
+
+Tests will be tagged as `[Feature:AppArmor]` like it is implemented right now,
+but they will be migrated to use the new fields API.
+
+New tests will be added covering the annotation/field conflict cases described
+under [Version Skew Strategy](#version-skew-strategy).
+
+### Graduation Criteria
+
+_This section is excluded, as it is the subject of the entire proposal._
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature enablement and rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
+
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+- [X] Feature gate 
+  - Feature gate name: `AppArmor`
+  - Components depending on the feature gate:
+      - kube-apiserver
+      - kubelet
+
+###### Does enabling the feature change any default behavior?
+  
+  N/A - the feature is already enabled by default since Kubernetes v1.4.
```

Comment:
Since we are adding new fields to the Pod API, I think there needs to be at least 1 version where the fields exist but not enabled to ensure rollback safety. Not sure if that warrants a new feature gate specifically for the new fields though.

Reply:
> Example:
>     1. Upgrade to v1.26: AppArmor enabled with new fields in Pod API
>     2. Rollback to v1.25: Pods (and APIs using PodTemplate) now have a field set that is not recognizable

I believe both of these relate to a similar conversation had [here](https://github.com/kubernetes/enhancements/blob/f771671feb06c9b130d5125526525096b9b7ff40/keps/sig-node/24-apparmor/README.md#upgrade--downgrade).

Copying fields to annotations and vice versa should account for bridging the gap during the supported version skew period, if I understand correctly.